chain/network/src/types.rs

```diff
@@ -525,27 +528,155 @@ impl Message for PeerRequest {
+    NetworkRequests(NetworkRequests),
+    Consolidate(Consolidate),
+    PeersRequest(PeersRequest),
+    PeersResponse(PeersResponse),
+    PeerRequest(PeerRequest),
+    GetPeerId(GetPeerId),
+    OutboundTcpConnect(OutboundTcpConnect),
+    InboundTcpConnect(InboundTcpConnect),
+    Unregister(Unregister),
+    Ban(Ban),
+    #[cfg(feature = "test_features")]
+    #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+    StartRoutingTableSync(StartRoutingTableSync),
+    #[cfg(feature = "test_features")]
+    SetAdvOptions(SetAdvOptions),
+    #[cfg(feature = "test_features")]
+    #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+    SetRoutingTable(SetRoutingTable),
+    GetRoutingTable(GetRoutingTable),
+}
+
+impl PeerMessageRequest {
+    pub fn as_network_requests(self) -> NetworkRequests {
+        if let PeerMessageRequest::NetworkRequests(item) = self {
+            item
+        } else {
+            panic!("expected PeerMessageRequest::NetworkRequests(");
+        }
+    }
+
+    pub fn as_network_requests_ref(&self) -> &NetworkRequests {
+        if let PeerMessageRequest::NetworkRequests(item) = self {
+            item
+        } else {
+            panic!("expected PeerMessageRequest::NetworkRequests(");
+        }
+    }
+}
+
+impl Message for PeerMessageRequest {
+    type Result = PeerMessageResponse;
+}
+
+#[derive(MessageResponse, Debug)]
+pub enum PeerMessageResponse {
+    RoutedMessageFrom(bool),
+    NetworkResponses(NetworkResponses),
+    ConsolidateResponse(ConsolidateResponse),
+    PeerRequestResult(PeerRequestResult),
+    PeersResponseResult(()),
+    PeerResponse(PeerResponse),
+    GetPeerIdResult(GetPeerIdResult),
+    OutboundTcpConnect(()),
+    InboundTcpConnect(()),
+    Unregister(()),
+    Ban(()),
+    #[cfg(feature = "test_features")]
+    #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+    StartRoutingTableSync(()),
+    #[cfg(feature = "test_features")]
+    SetAdvOptions(()),
+    #[cfg(feature = "test_features")]
+    #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+    SetRoutingTable(()),
+    GetRoutingTableResult(GetRoutingTableResult),
+}
+
+impl PeerMessageResponse {
+    pub fn as_routed_message_from(self) -> bool {
+        if let PeerMessageResponse::RoutedMessageFrom(item) = self {
+            item
+        } else {
+            panic!("expected PeerMessageRequest::RoutedMessageFrom(");
+        }
+    }
+
+    pub fn as_network_response(self) -> NetworkResponses {
+        if let PeerMessageResponse::NetworkResponses(item) = self {
+            item
+        } else {
+            panic!("expected PeerMessageRequest::NetworkRequests(");
+        }
+    }
+
+    pub fn as_consolidate_response(self) -> ConsolidateResponse {
+        if let PeerMessageResponse::ConsolidateResponse(item) = self {
+            item
+        } else {
+            panic!("expected PeerMessageRequest::NetworkRequests(");
```

Comment:
btw - you didn't update the contents of these panic (as this one should be about ConsolidateResponse)

Reply:
Fixed